applications/StructuralMechanicsApplication/custom_utilities/multipoint_constraint_data.hpp

```diff
@@ -0,0 +1,246 @@
+KratosMultiScaleApplication
+A library based on:
+Kratos
+A General Purpose Software for Multi-Physics Finite Element Analysis
+Version 1.0 (Released on march 05, 2007).
+
+Copyright 2007
+Pooyan Dadvand, Riccardo Rossi, 
+pooyan@cimne.upc.edu
+rrossi@cimne.upc.edu
+
+- CIMNE (International Center for Numerical Methods in Engineering),
+Gran Capita' s/n, 08034 Barcelona, Spain
+- Ruhr-University Bochum, Institute for Structural Mechanics, Germany
+
+
+Permission is hereby granted, free  of charge, to any person obtaining
+a  copy  of this  software  and  associated  documentation files  (the
+"Software"), to  deal in  the Software without  restriction, including
+without limitation  the rights to  use, copy, modify,  merge, publish,
+distribute,  sublicense and/or  sell copies  of the  Software,  and to
+permit persons to whom the Software  is furnished to do so, subject to
+the following condition:
+
+Distribution of this code for  any  commercial purpose  is permissible
+ONLY BY DIRECT ARRANGEMENT WITH THE COPYRIGHT OWNERS.
+
+The  above  copyright  notice  and  this permission  notice  shall  be
+included in all copies or substantial portions of the Software.
+
+THE  SOFTWARE IS  PROVIDED  "AS  IS", WITHOUT  WARRANTY  OF ANY  KIND,
+EXPRESS OR  IMPLIED, INCLUDING  BUT NOT LIMITED  TO THE  WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT  SHALL THE AUTHORS OR COPYRIGHT HOLDERS  BE LIABLE FOR ANY
+CLAIM, DAMAGES OR  OTHER LIABILITY, WHETHER IN AN  ACTION OF CONTRACT,
+TORT  OR OTHERWISE, ARISING  FROM, OUT  OF OR  IN CONNECTION  WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+==============================================================================
+*/
+//
+//   Project Name:        Kratos
+//   Last Modified by:    $Author: Aditya Ghantasala $
+//   Date:                $Date: 14-03-2017 $
+//   Revision:            $Revision: 1.00 $
+//
+//
+
+#if !defined(CONSTRAINT_SLAVE_H_INCLUDED)
+#define CONSTRAINT_SLAVE_H_INCLUDED
+
+#include <vector>
+#include <map>
+#include <iostream>
+
+namespace Kratos
+{
+
+/** \brief Quaternion
+	* A simple class that implements the main features of quaternion algebra
+	*/
+class MpcData : public Flags
+{
+
+  public:
+    typedef Dof<double>* DofPointerType;
+    typedef Dof<double> DofType;
+
+    ///@name Life Cycle
+    ///@{
+
+    /**
+		Creates a Quaternion from its coefficients.
+		@param id : The ID of the DOF which is considered as a slave
+		*/
+    MpcData()
+    {
+        this->totalNumberOfMasterDOFs = 0;
+    }
+    /// Destructor.
+    virtual ~MpcData(){};
+
+    ///@}
+
+  public:
+    ///@name Operators
+    ///@{
+
+    ///@}
+
+  public:
+    ///@name Access
+    ///@{
+
+    /**
+		Adds a master and a corresponding weight to the current slave
+		@param slaveDOF the DOF number which is to be added
+		*/
+    void AddSlave(DofPointerType slaveDOF)
+    {
+
+        if (this->slaveDofMasterDataMap.count(slaveDOF) > 0)

```

Comment:
I think the performance is the same, I have no objection in doing it in this way.